### ISSUE TITLE
Make dynamic_ncols=True the default

### ISSUE THREAD

**Issue by anntzer** [tqdm/tqdm#370]
Any reason why `dynamic_ncols` is not True by default?

- [X] I have visited the [source website], and in particular
  read the [known issues]
- [X] I have searched through the [issue tracker] for duplicates

  [source website]: https://github.com/tqdm/tqdm/
  [known issues]: https://github.com/tqdm/tqdm/#help
  [issue tracker]: https://github.com/tqdm/tqdm/issues?q=

**Comment by casperdcl** [comment]
could slow things down to constantly check terminal width. The ethos of `tqdm` is to be as fast as possible with default parameters, and to make it easy to make it even prettier at minimal cost by tweaking parameters. If there are enough votes to change a default, we'd be happy to do so.

**Comment by achalddave** [comment]
Sorry to comment on a closed issue, but is this the right place to vote for this? I frequently run into this issue when working with long-running python scripts in tmux panes. A common use case for me is that I ["zoom"](https://sanctum.geek.nz/arabesque/zooming-tmux-panes/) into a pane when running the script  and then zoom back out once I see a progress bar. Unfortunately, this doesn't work by default with tqdm, so my alternative is to ensure I add `dynamic_ncols=True` every time I use tqdm in any interactive script.

As far as I can tell from [here](https://github.com/tqdm/tqdm/issues/41), the overhead due to dynamic resizing is relatively small. However, if it cannot be added as a default, would it be possible to add an environment variable or a config that tqdm could check for to enable dynamic_ncols? This would let me avoid having to add `dynamic_ncols=True` to every incantation of tqdm in my scripts :)

**Comment by ThyrixYang** [comment]
I found it really annoying that I need to check the dynamic_ncols every time, I think there should be something dedicated for tmux use case like 
```python
from tqdm import dynamic_tqdm as tqdm
```
that would work for any legacy code.
If it's ok may I open a PR for it?

**Comment by casperdcl** [comment]
Could just add this to the top of your library:

```python
from functools import partial
from tqdm import tqdm as std_tqdm
tqdm = partial(std_tqdm, dynamic_ncols=True)
```

**Comment by ThyrixYang** [comment]
@casperdcl  Awesome! Thank you.

**Comment by casperdcl** [comment]
Thanks everyone for your patience! Just released [`tqdm==4.66.0`](https://github.com/tqdm/tqdm/releases/tag/v4.66.0) which includes support for env var overrides. I'm using `TQDM_MININTERVAL=5` in most of my CI jobs now :)

**Comment by nh2** [comment]
For this issue specifically, that's env var `TQDM_DYNAMIC_NCOLS=1`, right?
